docs/integrations/channels.md

```diff
@@ -137,11 +137,12 @@ class Subscription:
             )
 
         async for message in ws.channel_listen("chat.message", groups=room_ids):
-            yield ChatRoomMessage(
-                room_name=message["room_id"],
-                message=message["message"],
-                current_user=user,
-            )
+            if message["room_id"] in room_ids:
```

Comment:
So when you create a websocket on the client side, you can create multiple graphql "subscriptions" on one websocket, which will be handled by the same Consumer on the Django/Channels side. Each should get given a unique subscription ID. Because (in the channel layers backend) the concept of groups just subscribes the entire "Consumer" instance to the group, the async generators used by strawberry will see all the messages for all groups that the same websocket is subscribed to. I've tried to describe this in the documentation but happy to take comments if it could be clearer, as its not an easy thing to explain.

To try and illustrate it I put together a little demo from the example code in this document, and this is the browser dev tools showing the two subscriptions in one websocket, and you can see in my UI a message intended for one room shows up in both rooms (as I haven't included this condition in the example):

![image](https://user-images.githubusercontent.com/282776/218579125-8e23e9f8-847b-4581-abff-41de238aa6df.png)

The javascript client code is like this, note that I only instantiate one websocket client and re-use it for multiple calls to roomLogin()

```javascript
import { createClient } from "graphql-ws";

const subClient = createClient({ url: "ws://localhost:8000/graphql" })

export async function* subscribeToRoom(user, room) {
  let resolveMessage = () => { }

  subClient.subscribe({
    query: `
    subscription {
      joinChatRooms(user: "${user}", rooms:[ {roomName:"${room}"}]) {
        message,
        roomName
      }
    }
  `}, {
    next: (result) => {
      resolveMessage(result.data.joinChatRooms);
    }
  })

  while (true) {
    let nextMessage = new Promise((resolve, reject) => {
      resolveMessage = resolve;
    })
    yield await nextMessage;
  }
}

```